lib/ApiClient/AbstractCoreApiClient.php

```diff
@@ -145,4 +178,53 @@ protected function buildJsonParameters(array $required, array $optional)
 
         return $json;
     }
+
+    /**
+     * Generates the cache key based on request method, uri and options.
+     * @param string $method
+     * @param string $uri
+     * @param array $options
+     * @return string
+     */
+    private function generateCacheKey(string $method, string $uri, array $options = []): string
```

Comment:
Not sure if  I should remove the return type? I see now that PHP 5 should also be supported.